bundles/org.eclipse.ui.forms/src/org/eclipse/ui/forms/widgets/ExpandableComposite.java

```diff
@@ -1083,13 +1099,29 @@ public Color getTitleBarForeground() {
 
 	private void toggleState() {
 		boolean newState = !isExpanded();
-		fireExpanding(newState, true);
-		internalSetExpanded(newState);
-		fireExpanding(newState, false);
+		internalSetExpanded(newState, true);
 		if (newState)
 			FormUtil.ensureVisible(this);
 	}
 
+	/**
+	 * Performs the expansion state change for the expandable control.
+	 *
+	 * @param expanded
+	 *            the expansion state
+	 * @param fireExpanding
+	 *            if true, listeners will be fired
+	 */
+	private void internalSetExpanded(boolean expanded, boolean fireExpanding) {
+		if (fireExpanding)
+			fireExpanding(expanded, true);
+
+		internalSetExpanded(expanded);
+
+		if (fireExpanding)
+			fireExpanding(expanded, false);
```

Comment:
I'd would strongly suggest to always put expression blocks into curly braces.

Reply:
I also prefer it, I was just keeping the original style. Done.